### PR TITLE
fix(serializer): update json serializer signatures

### PR DESCRIPTION
**Pull request by benbenbang**
### **Description:**
As #166 stated, apply fixes for JsonXSerizer

### **Related Issue:**
#166 

### **Type of change**:

Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)

### **Checklist:**

- [x] All code follows the project's coding style and conventions.
- [x] I have performed a self-review of my own code.
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

File: confluent_kafka-stubs/schema_registry/json_schema.pyi [hunk 1]
```diff
@@ -25,7 +25,7 @@ class JSONSerializer(Serializer):
         self,
         schema_str: str | Schema,
         schema_registry_client: SchemaRegistryClient,
-        to_dict: Callable[[object], SerializationContext] | None = None,
+        to_dict: Callable[[dict[Any, Any], SerializationContext], object] | None = None,
```
Comment (calohmn) [inline confluent_kafka-stubs/schema_registry/json_schema.pyi:28]: The `to_dict` signature is different from the `from_dict` signature.

I think it should be
`to_dict: Callable[[object, SerializationContext], dict[Any, Any]] | None = None,`
(See also `avro.pyi`).

Comment (benbenbang) [reply confluent_kafka-stubs/schema_registry/json_schema.pyi]: True, thanks for taking a look!

Comment (benbenbang) [reply confluent_kafka-stubs/schema_registry/json_schema.pyi]: updated.

**Pull request merged**